api/src/main/java/org/jboss/cdi/tck/spi/Contexts.java

```diff
@@ -34,14 +34,22 @@
     public static final String PROPERTY_NAME = Contexts.class.getName();
 
     /**
-     * Sets the specified context as active
+     * Sets the specified context as active.
+     * <p>
+     * If the context was previously set as inactive via {@link  #setInactive(Context)}, this method is expected to
+     * restore the state of the context, including all contextual instances that were previously in it.
+     * <p>
+     * If the context wasn't previously set as inactive, a new context is created and activated.

```

Comment:
That can't be true. The `setActive` method accepts a context object, "a new context" can't be created.

Also, I'd avoid terms like "context state", as that's not a term defined by the specification. I believe something like this would be enough:

```java
/**
 * Sets the specified context as active.
 * <p>
 * The set of existing contextual instances of the context is preserved
 * across invocations of {@link #setActive(Context)} and {@link #setInactive(Context)}.
 *
 * @param context The context to set active
 */
public void setActive(T context);

/**
 * Sets the specified context as inactive.
 * <p>
 * The set of existing contextual instances of the context is preserved
 * across invocations of {@link #setActive(Context)} and {@link #setInactive(Context)}.
 *
 * @param context The context to set inactive
 */
public void setInactive(T context);
```

WDYT?

Reply:
> That can't be true. The setActive method accepts a context object, "a new context" can't be created.

You're right, that's poor wording. I meant to convey that the set of instances won't be there in any other case than when you previously set that context as inactive.

> * The set of existing contextual instances of the context is preserved
> * across invocations of {@link #setActive(Context)} and {@link #setInactive(Context)}.

Definitely better; but shouldn't we imply that the contextual instances are preserved only if you do `setInactive()` -> `setActive()`? Note that there is also destroy method plus you could attempt to invoke `setActive()` repeatedly after you destroy a contextual instance. But I guess I am just overthinking it :)

Reply:
Updated the text per your suggestion